devices/maas2/maas2.py

```diff
@@ -241,14 +255,14 @@ def deploy_node(self, distro="bionic", kernel=None, user_data=None):
         if user_data:
             data = base64.b64encode(user_data.encode()).decode()
             cmd.append("user_data={}".format(data))
-        process = subprocess.run(
-            cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+        proc = subprocess.run(
+            cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=False
         )
         try:
-            process.check_returncode()
+            proc.check_returncode()
```

Comment:
Is there some value to using check_returncode() and immediately catching the error over just checking the returncode member as happens everywhere else?

Reply:
yes, in this case we wanted the error output from it even if it failed. If we used check=True, then the exception would be generated and we wouldn't have a way to get the output.

Reply:
I meant:
```
if proc.returncode:
    log
    raise
```
not changing check=

Reply:
ah, thanks, I misunderstood. Yeah we can do that. Also improved it slightly to show the command in the log as well.